packages/fx-core/src/component/generator/generator.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ContextV3 } from "@microsoft/teamsfx-api";
+import path from "path";
+import {
+  Component,
+  sendTelemetryEvent,
+  TelemetryEvent,
+  TelemetryProperty,
+} from "../../common/telemetry";
+import { SampleActionSeq, GenerateAction, TemplateActionSeq } from "./generateAction";
+import { GenerateContext } from "./generateContext";
+import {
+  genFileDataRenderReplaceFn,
+  genFileNameRenderReplaceFn,
+  getSampleInfoFromName,
+  getValidSampleDestination,
+  mergeReplaceMap,
+  sampleDefaultOnActionError,
+  templateDefaultOnActionError,
+} from "./utils";
+
+export class Generator {
+  public static async generateTemplate(
+    templateName: string,
+    language: string,
+    destinationPath: string,
+    ctx: ContextV3
+  ): Promise<void> {
+    const appName = ctx.projectSetting?.appName;
+    const projectId = ctx.projectSetting?.projectId;
+    const dataReplace = mergeReplaceMap(
+      {
+        appName: appName,
+        projectId: projectId,
+      },
+      ctx.templateReplace
+    );
+    const nameReplace = mergeReplaceMap(
+      {
+        appName: appName,
+      },
+      ctx.templateReplace
+    );
+    const destination = ctx.isAddFeature ? destinationPath : path.join(destinationPath, appName);
```

Comment:
Do we actually create the folder called 'appName'?

Reply:
Yes. The create folder logic is at https://github.com/OfficeDev/TeamsFx/blob/5798adc3ec75b79d8672d503a02f5d680aed85cb/packages/fx-core/src/component/generator/utils.ts#L104